### PR TITLE
Refactor Affine Quantized Tensor

### PR DESCRIPTION
**Pull request by jainapurva**
Move Dtype Layout / TensorImpl class and dispatcher check-impl to their respective layout files
- Affine_quantized_tensor_ops.py contains 
     - aten.ops implementations for aqt
     - Dispatch registration and dtype kernels impl/check imports
- Dtype Layout files
  - torchao/dtypes/floatx
     - Float8_layout.py
     - Floatx_tensor_core_layout.py
  - torchao/dtypes/uintx
     - Block_sparse_layout.py
     - Marlin_sparse_layout.py
     - Plain_layout.py
     - Semi_sparse_layout.py
     - Tensor_core_tiled_layout.py
     - Uint4_layout.py
     - uintx_layout.py
     - marlin_qqq_layout.py

Test Plan: Pass all external and internal tests

### CODE REVIEW

File: torchao/dtypes/affine_quantized_tensor_ops.py [hunk 1]
```diff
@@ -0,0 +1,295 @@
+import logging
+from typing import Optional, Tuple
+
+import torch
+from torch.utils._python_dispatch import return_and_correct_aliasing
+from torchao.dtypes.affine_quantized_tensor import *
```
Comment (jerryzh168) [inline torchao/dtypes/affine_quantized_tensor_ops.py:6]: better to list the things you want to use I think

File: torchao/dtypes/uintx/uint8.py
Comment (jerryzh168) [inline torchao/dtypes/uintx/uint8.py:1]: uint8/uint8_layout are probably plain_layout?

File: torchao/dtypes/uintx/uintx.py
Comment (jerryzh168) [inline torchao/dtypes/uintx/uintx.py:1]: this is uintx_layout I think, we could also rename this later probably

File: torchao/dtypes/affine_quantized_tensor_ops.py [hunk 2]
```diff
@@ -0,0 +1,380 @@
...
+    raise QuantizedLinearNotImplementedError(
+        "No specialized dispatch found for quantized linear op"
+    )
+
+
+# Attach the _quantized_linear_op to the AffineQuantizedTensor class
+AffineQuantizedTensor._quantized_linear_op = _quantized_linear_op
+
+
+# # following are a list of (dispatch_condition, implementation) functions that takes the following args:
```
Comment (jerryzh168) [inline torchao/dtypes/affine_quantized_tensor_ops.py:104]: nit: probably need to remove one `#` from these lines

File: torchao/dtypes/uintx/uintx.py
Comment (jainapurva) [reply torchao/dtypes/uintx/uintx.py:1]: Yes, renamed to uintx_layout

File: torchao/dtypes/uintx/uint8.py
Comment (jainapurva) [reply torchao/dtypes/uintx/uint8.py:1]: Yes, renamed the file to plain_layout

File: torchao/dtypes/floatx/__init__.py [hunk 3]
```diff
@@ -12,4 +13,6 @@
     "to_scaled_tc_floatx",
     "from_scaled_tc_floatx",
     "_SPLIT_K_MAP",
+    "Float8AQTTensorImpl",
```
Comment (jerryzh168) [inline torchao/dtypes/floatx/__init__.py:16]: if this is not used by other people, maybe we can remove, this is supposed to be internal mostly I think

File: torchao/dtypes/uintx/__init__.py [hunk 4]
```diff
@@ -12,4 +31,11 @@
     "UintxAQTTensorImpl",
```
Comment (jerryzh168) [inline torchao/dtypes/uintx/__init__.py:31]: we can probably remove this one as well

File: torchao/quantization/quant_api.py [hunk 5]
```diff
@@ -36,7 +36,7 @@
     to_affine_quantized_floatx_static,
     to_affine_quantized_intx,
 )
-from torchao.dtypes.uintx.uintx import UintxLayout
+from torchao.dtypes.uintx.uintx_layout import UintxLayout
```
Comment (jerryzh168) [inline torchao/quantization/quant_api.py:39]: we can import from torchao.dtypes for now I think

File: torchao/dtypes/uintx/uint4_layout.py
Comment (jerryzh168) [inline torchao/dtypes/uintx/uint4_layout.py:1]: should be a follow up: this is not used actually, I think we can move this to prototype

Comment (jainapurva) [reply torchao/dtypes/uintx/uint4_layout.py:1]: Will do this in separate PR

File: torchao/dtypes/__init__.py [hunk 6]
```diff
@@ -16,15 +12,28 @@
...
+    SemiSparseLayout,
+    TensorCoreTiledLayout,
+    UintxLayout,
+)
+from .utils import (
+    Layout,
+    PlainLayout,
+)
 
 # from ..prototype.dtypes.uint2 import UInt2Tensor, BitnetTensor
```
Comment (jerryzh168) [inline torchao/dtypes/__init__.py:32]: same for  this line

File: test/hqq/test_hqq_affine.py [hunk 7]
```diff
@@ -1,12 +1,7 @@
 import unittest
 import torch
-from torchao.dtypes.affine_quantized_tensor import (
-    to_affine_quantized_intx,
+from torchao.quantization.quant_primitives import (
```
Comment (jerryzh168) [inline test/hqq/test_hqq_affine.py:3]: nit: maybe we can just import from torchao.quantization now

File: torchao/prototype/hqq/example.py [hunk 8]
```diff
@@ -2,13 +2,12 @@
...
 from torchao.dtypes.affine_quantized_tensor import (
     to_affine_quantized_intx,
-    ZeroPointDomain,
-    PlainAQTTensorImpl,
-    PlainLayout,
-    TensorCoreTiledAQTTensorImpl,
-    TensorCoreTiledLayout,
-    MappingType,
 )
+from torchao.quantization.quant_primitives import (
```
Comment (jerryzh168) [inline torchao/prototype/hqq/example.py:6]: same here

**Review by jerryzh168 — APPROVED** [review]
looks good, thanks! please import to fbcode and check if there is any internal failures

**Pull request merged**